src/Extension/CoreExtension.php

```diff
@@ -1642,7 +1642,7 @@ public static function getAttribute(Environment $env, Source $source, $object, $
 
         // array
         if (Template::METHOD_CALL !== $type) {
-            $arrayItem = \is_bool($item) || \is_float($item) ? (int) $item : $item;
+            $arrayItem = \is_bool($item) || \is_float($item) ? (int) $item : $item = (string) $item;
```

Comment:
The string casting breaks my ArrayAccess object using objects as keys, when using `myCollection[object]` in twig.

Reply:
Arf. Can you please submit a fix as a PR together with a test case?